tests/units/Services/Table/BuilderTest.php

```diff
@@ -30,235 +30,288 @@ public function setUp(): void
     {
-        $this->testModel->name = null;
-        $this->testModel->save();
+        $this->secondModel = $this->createTestModel();
 
-        $response = $this->requestResponse([
-            "columns" => [
-                "name" => ["data" => "name", "meta" => ["sortable" => true, "sort" => "ASC", "nullLast" => true]]
-            ],
-            'meta' => [
-                'sort' => true,
-            ]
-        ]);
+        $this->testModel->update(['name' => null]);
 
-        $this->assertNotNull($response["data"][0]["name"]);
+        $this->params['columns']['name'] = [
+            'data' => 'name',
+            'meta' => ['sortable' => true, 'sort' => 'ASC', 'nullLast' => true],
+        ];
+
+        $this->params['meta']['sort'] = true;
+
+        $response = $this->requestResponse();
+
+        $this->assertEquals(
+            $this->secondModel->name,
+            $response->get('data')->first()->get('name')
+        );
     }
 
     /** @test */
     public function can_get_data_with_cache()
     {
         Cache::shouldReceive('get')
-            ->andReturn(-10)
+            ->andReturn(12)
             ->shouldReceive('has')
             ->andReturn(true);
 
-        $response = $this->requestResponse([
-            'cache' => true
-        ]);
+        $this->params['cache'] = true;
 
-        $this->assertEquals(-10, $response["count"]);
+        $response = $this->requestResponse();
+
+        $this->assertEquals(12, $response->get('count'));
     }
 
     /** @test */
     public function can_get_data_with_limit()
     {
-        $response = $this->requestResponse([
-            'meta' => [
-                'length' => 0,
-            ]
-        ]);
+        $this->params['meta']['length'] = 0;
 
-        $this->assertCount(0, $response["data"]);
-    }
+        $response = $this->requestResponse();
 
+        $this->assertCount(0, $response['data']);
+    }
 
     /** @test */
     public function can_get_data_with_total()
     {
-        $response = $this->requestResponse([
-            "columns" => [
-                "id" => ['name' => 'id', "data" => "id", "meta" => ['total' => true]]
-            ],
-            'meta' => [
-                'total' => true,
-            ]
-        ]);
+        $this->createTestModel();
+
+        $this->params['columns']['price'] = [
+            'name' => 'price',
+            'data' => 'price',
+            'meta' => ['total' => true],
+        ];
+
+        $this->params['meta']['total'] = true;
+
+        $response = $this->requestResponse();
 
         $this->assertEquals(
-            BuilderTestModel::all()->sum("id"),
-            $response["total"]['id']
+            BuilderTestModel::sum('price'),
+            $response->get('total')->get('price')
         );
     }
 
-    private function requestResponse(array $params = [], $select = 'id as dtRowId,name,is_active,created_at,price', $withFetch = false)
+    public function can_use_full_info_records_limit()
```

Comment:
It seems this method could be deleted. Is it right?

Reply:
why?

Reply:
the method is 
```
public function can_use_full_info_records_limit()
    {
        $this->createTestModel();

        $this->params['meta']['fullInfoRecordLimit'] = 1;
        $this->params['meta']['length'] = 1;

        $response = $this->requestResponse();

        $this->assertEquals(
            BuilderTestModel::sum('price'),
            $response->get('total')->get('price')
        );
    }```

it's not test and logic seems to be wrong

Reply:
Ah totally. I was looking below, addressed just now